js/tidelinedata.js

```diff
@@ -364,7 +364,12 @@ function TidelineData(data, opts) {
           }
           // timezoneOffset is an optional attribute according to the Tidepool data model
           else {
-            d.normalTime = d.deviceTime + '.000Z';
+            if (_.isEmpty(d.deviceTime)) { 
+               d.normalTime = d.time;
+            }
+            else {
+               d.normalTime = d.deviceTime + '.000Z';
+            }
             d.displayOffset = 0;
```

Comment:
If you are going to factor out the `d.displayOffset = 0;` from these two clauses, you might want to do it for all of the clauses in the if/else statement. Might make the if/else if/else if/else statement cleaner.